test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs

```diff
@@ -333,7 +348,59 @@ public void InvalidUser()
             Assert.NotNull(authorizationContext.Result);
         }
 
-        private Filters.AuthorizationFilterContext GetAuthorizationContext(
+        [Fact]
+        public void CreateInstance_ReturnsSelfIfPolicyIsSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new AuthorizationPolicyBuilder()
+                .RequireAssertion(_ => true)
+                .Build());
+            var factory = (IFilterFactory)authorizeFilter;
+
+            // Act
+            var result = factory.CreateInstance(new ServiceCollection().BuildServiceProvider());
+
+            // Assert
+            Assert.Same(authorizeFilter, result);
+        }
+
+        [Fact]
+        public void CreateInstance_ReturnsSelfIfPolicyProviderIsSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new AuthorizationPolicyBuilder()
+                .RequireAssertion(_ => true)
+                .Build());
+            var factory = (IFilterFactory)authorizeFilter;
+
+            // Act
+            var result = factory.CreateInstance(new ServiceCollection().BuildServiceProvider());
+
+            // Assert
+            Assert.Same(authorizeFilter, result);
+        }
+
+        [Fact]
+        public void CreateInstance_ReturnsNewPolicyIfPolicyAndPolicyProviderIsNotSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new[] { new AuthorizeAttribute() });
+            var factory = (IFilterFactory)authorizeFilter;
+            var serviceProvider = new ServiceCollection()
+                .AddOptions()
+                .AddAuthorization()
```

Comment:
could you add a test/variation where you supply custom authorization policy provider